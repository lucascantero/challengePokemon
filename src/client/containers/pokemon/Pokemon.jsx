import React, { useEffect, useState } from 'react';
import { useParams } from "react-router";
import Header from '../../components/header/Header';
import { useTranslation } from 'react-i18next';

const Pokemon = () => {

    const [data, setData] = useState({ abilities: [] });
    const { t } = useTranslation();

    let { id } = useParams();

    useEffect(() => {

        const getData = async () => {
            const response = await fetch(`/api/pokemon/${id}`);
            const dataAPI = await response.json();
            setData(dataAPI);
        }
        getData();
    }, []);

    return (
        <>
            <Header />
            <div className="pokemon__container">
                <div className="pokemon__wrapper">
                    <div className="pokemon__wrapper-catalogo">
                        <div className="pokemon__wrapper-img-title">
                            <div className="pokemon__wrapper-img">
                                <img src={data.img} className="pokemon__img" />
                            </div>
                            <div className="pokemon__wrapper-title">
                                <h1 className="pokemon__name">
                                    {data.name}
                                </h1>
                                {t("abilities")}:
                                <div className="pokemon__abailities">                           
                                {
                                    data.abilities.map(ability => (
                                        <h2 key={ability} className="pokemon__abaility"style={{padding:'10px'}}> 
                                            {ability}
                                        </h2>
                                    ))
                                }
                                </div>
                                <h3 className="pokemon__description">
                                    {t("description")}: {data.description}
                                </h3>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </>
    )
}

export default Pokemon;