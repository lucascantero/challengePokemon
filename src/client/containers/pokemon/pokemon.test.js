import React from 'react';
import Pokemon from './Pokemon.jsx';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';

describe('Testing Product', () => {

    let container = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it("Should match snapshot", async () => {

        const fakeUser = { 
                "name": "name",
                "id": "id",
                "description": "description",
                "abilities": ["abilities"],
                "img": "img"
        }

        global.fetch = jest.fn().mockImplementation
            (() =>
                Promise.resolve({
                    json: (() => Promise.resolve(fakeUser))
                })
            );

        await act(async () => {
            return render(<MemoryRouter> <Pokemon /> </MemoryRouter>, container);
        });

        expect(container.innerHTML).toMatchSnapshot();

    });

})