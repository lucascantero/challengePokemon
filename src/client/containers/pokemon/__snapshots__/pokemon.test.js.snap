// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Product Should match snapshot 1`] = `" <div class=\\"searchBox__contenedor\\"><div class=\\"buttonLanguage__wrapper\\"><button class=\\"buttonLanguage__button\\">en</button><button class=\\"buttonLanguage__button\\">es</button></div></div><div class=\\"pokemon__container\\"><div class=\\"pokemon__wrapper\\"><div class=\\"pokemon__wrapper-catalogo\\"><div class=\\"pokemon__wrapper-img-title\\"><div class=\\"pokemon__wrapper-img\\"><img class=\\"pokemon__img\\" src=\\"img\\"></div><div class=\\"pokemon__wrapper-title\\"><h1 class=\\"pokemon__name\\">name</h1>abilities:<div class=\\"pokemon__abailities\\"><h2 class=\\"pokemon__abaility\\" style=\\"padding: 10px;\\">abilities</h2></div><h3 class=\\"pokemon__description\\">description: description</h3></div></div></div></div></div> "`;
